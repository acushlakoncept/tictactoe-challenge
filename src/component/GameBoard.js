/* eslint-disable react/no-array-index-key */
import PropTypes from 'prop-types';
import Square from './Square';

const GameBoard = ({ squares, handleClick }) => (
  <div className="gameBoard">
    {squares.map((square, i) => (
      <Square key={i} value={square} handleClick={() => handleClick(i)} />
    ))}
  </div>
);

GameBoard.propTypes = {
  squares: PropTypes.arrayOf.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default GameBoard;
