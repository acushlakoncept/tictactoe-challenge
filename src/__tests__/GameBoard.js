import { func } from 'prop-types';
import React from 'react';
import renderer from 'react-test-renderer';
import GameBoard from '../component/GameBoard';

it('renders GameBoard correctly', () => {
  const cells = Array(9).fill(null);
  const tree = renderer.create(<GameBoard squares={cells} handleClick={func} />);
  expect(tree).toMatchSnapshot();
});
