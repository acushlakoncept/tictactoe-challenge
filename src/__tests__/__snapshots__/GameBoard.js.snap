// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GameBoard correctly 1`] = `
<div
  className="gameBoard"
>
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
</div>
`;

exports[`renders StockCard correctly 1`] = `
<div
  className="gameBoard"
>
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
  <button
    className="square btn-light null btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  />
</div>
`;
